backend/src/memberships/memberships.controller.ts

```diff
@@ -9,27 +9,121 @@ import {
   BadRequestException,
   Query,
   NotFoundException,
+  UnauthorizedException,
+  Req,
 } from '@nestjs/common';
 import { MembershipsService } from './memberships.service';
 import {
   UpdateMembershipDto,
   CreateMembershipDto,
   ResponseMembershipDto,
   QueryMembershipDto,
+  MembershipRoleType,
 } from '@dtos/memberships';
 import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
 import { EntityDoesNotExistError } from 'src/errors/entityDoesNotExist';
+import { ChannelsService } from 'src/channels/channels.service';
+import { ChannelType } from 'src/channels/entities/channel.entity';
+import { EntityNotFoundError } from 'typeorm';
+import * as bcrypt from 'bcrypt';
+import { Request } from 'express';
+import { User } from 'src/users/entities/user.entity';
+
+async function userIsAdmin(
+  userId: number,
+  channelId: number,
+  membershipsService: MembershipsService,
+): Promise<boolean> {
+  const creatorMembership = await membershipsService.findAll({
+    user: userId.toString(),
+    channel: channelId.toString(),
+  });
+  if (
```

Comment:
I thought there could be only one owner but several admin, am I mistaken ? What if we find several matches, does that mean no one is admin ?

Reply:
You are correct, there can only be one owner. Here we are checking to see what a given user's role in a channel is, if we find several matches it means that there is a problem with the database or the query (because a user should not have multiple memberships to the same channel), but we could possibly find 0 matches, which would mean that this user is not a member of the channel at all

It might be a good idea to add a unit test for that, to make sure this is working properly

Reply:
Thank you for the explanation !